### PR TITLE
Add reStructuredText converter

### PR DESCRIPTION
**Pull request by ikalnitsky**

### CODE REVIEW

File: docs/quickstart.rst [hunk 1]
```diff
@@ -94,11 +94,11 @@ type. Here's the key:
...
 **about.mdown**
   A sample page. Each file that could be parsed by one of enabled converters
-  (default: Markdown only) is considered as either Page or Post. Both Page
-  and Post documents preserve their relative paths to the content directory
-  and will be available under almost the same path (a file extension will be
-  stripped) in the output one. I.e. ``$content/a/b/page.mdown`` will be
-  available under the ``$host/a/b/page/`` url.
+  is considered as either Page or Post. Both Page and Post documents preserve
+  their relative paths to the content directory and will be available under
+  almost the same path (a file extension will be stripped) in the output one.
```
Comment (PropheticBird) [inline docs/quickstart.rst]: better to say 'directory' instead of 'one' in this case

File: docs/quickstart.rst [hunk 2]
```diff
@@ -115,7 +115,7 @@ type. Here's the key:
 What all that means? That means
 
 * If you want to add new page, just drop it anywhere in the content folder
-  in the supported format (default: Markdown only).
+  in the supported format (default: Markdown & reStructuredText only).
```
Comment (PropheticBird) [inline docs/quickstart.rst]: perhaps we don't need `only` anymore

Comment (ikalnitsky) [inline docs/quickstart.rst]: Fixed in local branch. Will be updated.

File: docs/quickstart.rst (hunk 1 above)
Comment (ikalnitsky) [inline docs/quickstart.rst]: Fixed in local branch. Will be updated.

File: holocron/ext/converters/restructuredtext.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    if you know it deep enough (it's hard) you can pass custom settings
+    using the following option::
+
+        ext:
+           restructuredtext:
+              docutils: { ... }
+
+    The class is actually both extension and converter in terms of Holocron
+    at one time. It means that this class will be discovered by Holocron as
+    extension, and this class register its instance as converter in the
```
Comment (PropheticBird) [inline holocron/ext/converters/restructuredtext.py]: and this class register`s` its instance as `a` converter

File: holocron/ext/converters/restructuredtext.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    }
+
+    def __init__(self, app):
+        self._conf = Conf(
+            self._default_conf,
+            {
+                'input_encoding': app.conf['encoding.content'],
+                'output_encoding': app.conf['encoding.output'],
+            },
+            app.conf.get('ext.restructuredtext', {}))
```
Comment (PropheticBird) [inline holocron/ext/converters/restructuredtext.py]: maybe move `{}))` to the next line?

File: tests/ext/converters/test_markdown.py [hunk 5]
```diff
@@ -44,7 +44,33 @@ def test_markdown_simple_post(self):
         self.assertEqual(meta['title'], 'some title')
         self.assertEqual(html, '<p>some text with <strong>bold</strong></p>')
 
-    def test_markdown_without_title(self):
+    def test_post_with_sections(self):
+        """
+        Title must gone, section must be converted into <h2>.
```
Comment (PropheticBird) [inline tests/ext/converters/test_markdown.py]: must `be` gone

File: tests/ext/converters/test_restructuredtext.py [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+        meta, html = self.conv.to_html(
+            'some text with **bold** and *italic*\n')
+
+        self.assertNotIn('title', meta)
+        self.assertEqual(html, (
+            '<p>some text with <strong>bold</strong> and <em>italic</em></p>'))
+
+    def test_post_with_sections(self):
+        """
+        Title must gone, section must be converted into <h2>.
```
Comment (PropheticBird) [inline tests/ext/converters/test_restructuredtext.py]: same as above

**Comment by PropheticBird** [comment]
Could you also update `holocron/example/_config.yml` since we include this file in the documentation?

File: holocron/ext/converters/restructuredtext.py (hunk 3 above)
Comment (ikalnitsky) [inline holocron/ext/converters/restructuredtext.py]: Fixed.

File: holocron/ext/converters/restructuredtext.py (hunk 4 above)
Comment (ikalnitsky) [inline holocron/ext/converters/restructuredtext.py]: For what purpose? `{})` is a default value. I'd prefer to keep it as is. You know, I just pass arguments one-by-one to `Conf` constructor.

File: tests/ext/converters/test_markdown.py (hunk 5 above)
Comment (ikalnitsky) [inline tests/ext/converters/test_markdown.py]: Fixed. Thank you.

**Comment by ikalnitsky** [comment]
> Could you also update holocron/example/_config.yml since we include this file in the documentation?

Nice catch! Thank you. I definitely will.

File: holocron/ext/converters/restructuredtext.py (hunk 4 above)
Comment (PropheticBird) [inline holocron/ext/converters/restructuredtext.py]: Oops, I thought, `{}` is related to `Conf` constructor. Ah, forget it :)

**Pull request merged**